### PR TITLE
Go to definition for type predicate

### LINKED ISSUE #3600: GoToDefinition doesn't seem to work on parameter of a type predicate
Might be an issue with how I did it, but I built TypeScript and patched up the tsserver.js in my Sublime directory. Then tried this out:

```TypeScript
function f(parameter: any): /**/parameter is string {
    return typeof parameter === "string";
}
```

Go-to-definition doesn't work on the LHS of the type predicate type.

### PR DESCRIPTION
**Pull request by tinganho**
Fixes #3600.

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -11925,6 +11925,10 @@ namespace ts {
                 return resolveEntityName(<EntityName>entityName, meaning);
             }
 
+            if (entityName.parent.kind === SyntaxKind.TypePredicate) {
+                return resolveEntityName(<Identifier>entityName, /* meanings */ SymbolFlags.FunctionScopedVariable);
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: nit: We usually don't put space around the name for the comment

Comment (tinganho) [inline src/compiler/checker.ts]: thanks, I also noticed that the parameter name is without plural. Fixed now.

**Comment by DanielRosenwasser** [comment]
:+1:

**Comment by mhegazy** [comment]
thanks!

**Pull request merged**